presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java

```diff
@@ -92,6 +92,8 @@ public ThriftHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, ThriftFie
 
         return Optional.of(new ThriftHiveRecordCursor<>(
                 genericRecordReader(recordReader),
+                path,
+                start,
```

Comment:
Here is just creating the record cursor and not the place to actually read the records. If we want to display the detailed split path and range, we need to pass these two parameters into the cursor. And add them into the error message in the "advanceNextPosition" method, which is the public method for interface `com.facebook.presto.spi.RecordCursor` and called by the `getNextPage` in the `com.facebook.presto.spi.RecordPageSource`